### ISSUE TITLE
Comments not added when creating DHCP-lease

### ISSUE THREAD

**Issue by TheBeems** [ddelnano/terraform-provider-mikrotik#28]
First of all great project with the terraform provider for both xen and mikrotik. My homelab uses both, so fun to play with!

Found a minor bug in the comment section when creating a dhcp-lease. When adding a comment with spaces, only the first word is being added in the mikrotik comment section. This is because the comment isn't between " ". See terraform debug log below. After it is added, everytime you do a terraform plan/apply it wants to change the comment again because the terraform state and the real value in the mikrotik router differ.

> module.testing.mikrotik_dhcp_lease.lease: Modifying... [id=*12FE6]
> 2021-02-20T09:13:39.475+0100 [DEBUG] plugin.terraform-provider-mikrotik_v0.5.0.exe: 2021/02/20 09:13:39 [INFO] Running the mikrotik command: **`[/ip/dhcp-server/lease/set =.id=*12FE6 =address=172.30.42.12 =mac-address=AA:BB:CC:DD:EE:12 =comment=A VM without a purpose =block-access=false]`**
> 2021-02-20T09:13:39.501+0100 [DEBUG] plugin.terraform-provider-mikrotik_v0.5.0.exe: 2021/02/20 09:13:39 [INFO] Running the mikrotik command: `[/ip/dhcp-server/lease/print ?.id=*12FE6]`
> 2021-02-20T09:13:39.512+0100 [DEBUG] plugin.terraform-provider-mikrotik_v0.5.0.exe: 2021/02/20 09:13:39 [DEBUG] Dhcp lease response: !re @ [{`.id` `*12FE6`} {`address` `172.30.42.12`} {`mac-address` `AA:BB:CC:DD:EE:12`} {`address-lists` ``} {`server` `defconf`} {`dhcp-option` ``} {`status` `bound`} {`expires-after` `9m33s`} {`last-seen` `27s`} {`active-address` `172.30.42.12`} {`active-mac-address` `AA:BB:CC:DD:EE:12`} {`active-client-id` `ff:cc:dd:ee:12:0:1:0:1:27:bd:f:a0:d6:3d:5e:b5:eb:1d`} {`active-server` `defconf`} {`host-name` `testing`} {`radius` `false`} {`dynamic` `false`} {`blocked` `false`} {`disabled` `false`} **{`comment` `A`}**]
> 2021-02-20T09:13:39.512+0100 [DEBUG] plugin.terraform-provider-mikrotik_v0.5.0.exe: !done @ []
> 2021/02/20 09:13:39 [WARN] Provider "registry.terraform.io/ddelnano/mikrotik" produced an unexpected new value for module.testing.mikrotik_dhcp_lease.lease, but we are tolerating it because it is using the legacy plugin SDK.
>     **The following problems may be the cause of any confusing errors from downstream operations:
>       - .comment: was cty.StringVal("A VM without a purpose"), but now cty.StringVal("A")**
> 2021-02-20T09:13:39.537+0100 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
> 2021-02-20T09:13:39.555+0100 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/ddelnano/mikrotik/0.5.0/windows_amd64/terraform-provider-mikrotik_v0.5.0.exe pid=11608
> m2021-02-20T09:13:39.571+0100 [DEBUG] plugin: plugin exited
> odule.testing.mikrotik_dhcp_lease.lease: Modifications complete after 1s [id=*12FE6]

**Comment by TheBeems** [comment]
I have no experience in programming in Go, but checked out the documentation (https://golang.org/pkg/fmt/):

So maybe changing `comment=%s` to `comment=%q` at the following line at should fix it:

https://github.com/ddelnano/terraform-provider-mikrotik/blob/b1206467a9736dbeb90dcc0b6a2963ab7c9a55f8/client/lease.go#L25

I have no idea how to change or test this myself or submit to github. But maybe the above information saves you the hassle to find a solution yourself.

**Comment by ddelnano** [comment]
> First of all great project with the terraform provider for both xen and mikrotik. My homelab uses both, so fun to play with!

Glad you are getting use out of it. It's what I use for my homelab too and appreciate the detailed bug report!

I gave your theory a try (using the `%q`) and unfortunately it doesn't seem to be enough :(. I added a failing test case on [this branch](https://github.com/ddelnano/terraform-provider-mikrotik/commit/4f1959392a13d990ca37f70b74ee103969b35da8) and here is the output of the Terraform debug logs during a failed test. It shows the request is correctly passing the multi line string, however, the mikrotik API response is still returning the first word in the comment string.

I will need to look into this more. I'm not sure if the mikrotik command that is being passed to `RunArgs` is correct or not.

```
$ TF_LOG=debug TEST=TestAccMikrotikDhcpLease_create make testacc

...
...
...
2021/02/20 21:35:43 [INFO] Running the mikrotik command: `[/ip/dhcp-server/lease/add =address=1.1.1.1 =mac-address=11:11:11:11:11:11 =comment="multi word comment" =block-access=false]`
2021/02/20 21:35:43 [DEBUG] Dhcp lease creation response: `!done @ [{`ret` `*50EF`}]`
2021/02/20 21:35:43 [INFO] Running the mikrotik command: `[/ip/dhcp-server/lease/print ?.id=*50EF]`
2021/02/20 21:35:43 [DEBUG] Dhcp lease response: !re @ [{`.id` `*50EF`} {`address` `1.1.1.1`} {`mac-address` `11:11:11:11:11:11`} {`address-lists` ``} {`dhcp-option` ``} {`status` `waiting`} {`last-seen` `never`} {`radius` `false`} {`dynamic` `false`} {`blocked` `false`} {`disabled` `false`} {`comment` `"multi`}]
```

**Comment by ddelnano** [comment]
Running this command from the mikrotik shell seems to work

```
[admin@MikroTik] /ip dhcp-server lease> add mac-address=11:11:11:11:11:11 address=1.1.1.1 comment="Testing one two"
[admin@MikroTik] /ip dhcp-server lease> print
Flags: X - disabled, R - radius, D - dynamic, B - blocked
 #   ADDRESS                                                               MAC-ADDRESS       HOST-NAME                             SERVER                             RATE-LIMIT                             STATUS
...
...

34 D 192.168.88.34                                                         9C:F3:87:D0:F5:10 CIs-MacBook-Air                       defconf                                                                   bound
35   ;;; Testing one two
     1.1.1.1                                                               11:11:11:11:11:11                                                                                                                 waiting
```

However, as you can tell there is a slightly different syntax between what is supported on the mikrotik cli and the input + encdoing that the mikrotik go client does.

So I think this is an issue with how the client is "encoding" the request.

**Comment by ddelnano** [comment]
Yep, the `strings.Split` was the problem here. This will be fixed in v0.5.1 of the provider which I'm releasing now.

**Comment by ddelnano** [comment]
There are likely bugs with other fields in the provider which I will plan to tackle more generically in #30.
